be/src/vec/functions/function_timestamp.cpp

```diff
@@ -52,8 +52,60 @@ struct StrToDate {
             if (!ts_val.from_date_format_str(r_raw_str, r_str_size, l_raw_str, l_str_size)) {
                 null_map[i] = 1;
             }
-            ts_val.to_datetime();
+            if (context->impl()->get_return_type().type ==
+                doris_udf::FunctionContext::Type::TYPE_DATETIME) {
+                ts_val.to_datetime();
+            } else {
+                ts_val.cast_to_date();
+            }
+        }
+    }
+};
+
+template <typename Impl>
+class FunctionStrToDateOrDateTime : public IFunction {
```

Comment:
should change other function of `FunctionBinaryStringOperateToNullType` and add `FunctionContext`

Reply:
done